flax/core/nn/attention.py

```diff
@@ -299,8 +299,8 @@ def init_fn(shape, dtype=jnp.float32):
         indices[attn_dim] = i // attn_size
         i = i % attn_size
 
-      key = lax.dynamic_update_slice(cache_entry.key, key, indices)
-      value = lax.dynamic_update_slice(cache_entry.value, value, indices)
+      key = lax.dynamic_update_slice(cache_entry.key, key, indices) # type: ignore
```

Comment:
just curious what type error lax.dynamic_update_slice causes?

Reply:
`flax/core/nn/attention.py:302: error: Argument 1 to "dynamic_update_slice" has incompatible type "ndarray[Any, Any]"; expected "Array"  [arg-type]
`

To do this correctly, I think I'd need to update the types for `CacheEntry` and that felt like too intrusive of a step.